sys/kern/kern_exec.c

```diff
@@ -1655,54 +1697,69 @@ exec_copyout_strings(struct image_params *imgp)
 		execpath_len = 0;
 	p = imgp->proc;
 	szsigcode = 0;
+
+#if __has_feature(capabilities)
+	/* XXX: should be derived from a capability to the strings region */
```

Comment:
I think we should be creating a capability to ps_strings and hanging it off struct proc somewhere in the process. I think that would help tidy of get_proc_vector() as well.

Reply:
Yes, probably.  I was just preserving that XXX comment from the cheriabi version.